### ISSUE TITLE
[Enhancement] keep list in memory

### ISSUE THREAD

**Issue by Efreak** [SmartPack/PackageManager#27]
When searching the package list, it appears to be fetching the list again before applying the filter; it would be nice if the full list was kept in memory, so searching could be instant. The same goes for switching between theAll/System/User tabs at the top.

**Comment by sunilpaulmathew** [comment]
You;re right. At this moment, whenever user switch between options, Package Manager will try to re-acquire the package details again (same thing happened while searching as well). Infact, app doesn't save anything to memory. The main issue is, even we save everything in memory, loading applications will take lot of time especially on devices with a lot of applications installed. It is because UI needs to load some basic details (App name, icon etc.) each time. A really don't know there is a better method exist.

**Comment by Efreak** [comment]
I know it's possible, but might not be worth the bother. Some apps (AFWall+) have an option to skip loading the icons in a list of apps; this certainly speeds things up. I'm not an Android/Java developer, but [this] (https://www.vogella.com/tutorials/AndroidListView/article.html#adapterperformance) might also help, possibly?
